Dockerfile

```diff
@@ -13,6 +17,12 @@ RUN bundle install --jobs 4 && npm install && npm install -g phantomjs-prebuilt
 
 EXPOSE 3000
 ADD . /myapp
-RUN RAILS_ENV=production bundle exec rake assets:precompile
+RUN if [ -z "$CUSTOM_ASSETS_URL" ]; then\
```

Comment:
You could also use environment variables instead of build arguments. E.g. extend the rake task the way Vince suggested, and check in the rake task if the necessary env variables have been set (e.g. `CUSTOM_ASSETS_PATH`) - if not, use the default assets, if yes, run all the logic in the rake task.